Apps/Inc/SendTritium.h

```diff
@@ -1,15 +1,24 @@
 #ifndef __SENDTRITIUM_H
 #define __SENDTRITIUM_H
 
+#include "getter.h"
+
 #define MOTOR_MSG_PERIOD 500
 #define FSM_PERIOD 250
 #define DEBOUNCE_PERIOD 2 // in units of FSM_PERIOD
 #define MOTOR_MSG_COUNTER_THRESHOLD (MOTOR_MSG_PERIOD)/(FSM_PERIOD)
 
-typedef enum{
-	FORWARD_GEAR,
-	NEUTRAL_GEAR,
-	REVERSE_GEAR
+#define FOREACH_Gear(GEAR) \
+        GEAR(FORWARD_GEAR)   \
+        GEAR(NEUTRAL_GEAR)  \
+        GEAR(REVERSE_GEAR)   \
+
+#define GENERATE_ENUM(ENUM) ENUM,
+#define GENERATE_STRING(STRING) #STRING,
+
+typedef enum GEAR_ENUM {
+    FOREACH_Gear(GENERATE_ENUM)
+    NUM_GEARS,
```

Comment:
The only reason for doing this macro is so that it can be printed out with the string. Without it, that couldn't be the case. I do agree that it looks complex though. I'll look into a better way to make it seem cleaner.